spec/controllers/resources_controller_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+DatabaseCleaner.strategy = :truncation
+
+describe ResourcesController, type: :request do
+
+  before do
+    Slosilo["authn:rspec"] ||= Slosilo::Key.new
+  end
+
+  let(:current_user) { Role.find_or_create(role_id: 'rspec:user:admin') }
+
+  describe '#post' do
+
+    before(:each) do
+      load_variables()
+    end
+
+    let(:resources_url) do
+      '/resources/rspec'
+    end
+
+    let(:policies_url) do
+      '/policies/rspec/policy/root'
+    end
+
+    let(:token_auth_header) do
+      bearer_token = Slosilo["authn:rspec"].signed_token(current_user.login)
+      token_auth_str =
+        "Token token=\"#{Base64.strict_encode64(bearer_token.to_json)}\""
+      { 'HTTP_AUTHORIZATION' => token_auth_str }
+    end
+
+    def count_resources_without_limit()
+      get(resources_url, env: token_auth_header(), params: {:count => "true"})
+    end
+
+    def count_resources_with_limit(limit)
+      get(resources_url, env: token_auth_header(), params: {:limit => limit, :count => "true"})
+    end
+
+    def load_variables()
+      payload = '[!variable preexisting\n!variable preexisting1]'
+      put(policies_url, env: token_auth_header.merge({ 'RAW_POST_DATA' => payload }))
+    end
+
+    context 'with default configuration' do
+      it "should count all resources when limit not defined" do
+        count_resources_without_limit()
+        expect(response.code).to eq("200")
+        expect(response.body).to eq("{\"count\":2}")
+      end
+
+      it "should count resources according to query param limit when limit defined" do
+        count_resources_with_limit(1)
+        expect(response.code).to eq("200")
+        expect(response.body).to eq("{\"count\":1}")
+      end
+    end
+
+    context 'with custom configuration' do
+      it "should count resources according to custom configuration when limit not defined" do
+        Rails.application.config.conjur_config.api_resource_list_limit = 1
+        count_resources_without_limit()
+        expect(response.code).to eq("200")
+        expect(response.body).to eq("{\"count\":1}")
+      end
+
+      it "should count resources according to query param limit when custom configuration exceeds limit" do
+        Rails.application.config.conjur_config.api_resource_list_limit = 2
+        count_resources_with_limit(1)
+        expect(response.code).to eq("200")
+        expect(response.body).to eq("{\"count\":1}")
+      end
+
+      it "should throw error when limit exceeds custom configuration" do
+        Rails.application.config.conjur_config.api_resource_list_limit = 1
+        count_resources_with_limit(2)
+        expect(response.code).to eq("422")
+      end
+    end
```

Comment:
This appears to introduce a bug. 

`limit` should never impact the count, even if they're inadvertently provided together. That's a gap in our validation now, but we shouldn't propagate the issue further. If `count` is true, we should not be applying the default limit value.

We do want the ability to get the total number of resources in Conjur that match the other parameters.

All of these test cases should be:
```ruby
expect(response.body).to eq("{\"count\":2}")
```

And there should be an additional error case that `limit` and `count` should not be used together.

Reply:
That also means we do need test cases that retrieve the records and not just count them.

Reply:
@aloncarmel111, I'm glad we're finally rolling out a limit on resources returned as a single result.  It's a very common pattern across APIs to solve the issue the security team identified. 

Instead of setting the default to `0` (no limit), I feel it makes sense to set it to something reasonable for everyone (ex. `100`). I don't love the idea of making it configurable, but it does allow customers who depend on getting all resources in a single request to continue to support this behavior (while understanding the downside of this decision).

Reply:
We can align to other CyberArk, in PVWA the default limit is 50. and the maximum number that can be specified is 1000.
@jvanderhoof wdyt the expected behavior should be when `limit` and `count` used together?

Reply:
@micahlee , Updated the tests to retrieve the records and not just count them.
I will create issue to decide the expected behavior when `limit` and `count` used together

Reply:
Hi @micahlee, added validation for not applying the default limit value when **count** set to true 
also opened the following issue https://github.com/cyberark/conjur/issues/2579 
can you prioritize this?

Reply:
@jvanderhoof, The maximum limit for Conjur cloud is set to 50. DAP and OSS are in other scale than Conjur cloud, I will open issue for this, we need to understand the exact limit for DAP and OSS,  because it's can impact existing customer.
CC @rrefael @netacoral

Reply:
@aloncarmel111, if `limit` and `count` are used together, I'd expect `limit` to have no effect, as a single response is returned (the total number of that resource search).